source/settings/plugin/quota-clone-plugin.rst

```diff
@@ -11,28 +11,33 @@ quota-clone plugin
 Settings
 ========
 
-.. dovecot_plugin:setting:: quota_clone_dict
+.. dovecot_plugin:setting_filter:: quota_clone
+   :filter: quota_clone
    :plugin: quota-clone
-   :values: @string
+   :setting: dict_driver
+   :values: @named_filter
+   :added: 2.4.0,3.0.0
 
-   The dictionary to update with quota clone information. This must be set for
-   the plugin to be active.
+   Named filter for initializing dictionary used to update with quota clone
+   information.
 
    See :ref:`dict` for dictionary configuration.
 
-   Example:
+   Options:
 
-   .. code-block:: none
+   ``unset``
 
-     plugin {
-       quota_clone_dict = redis:host=127.0.0.1:port=6379
-     }
+      Unset quota information before updating. This is needed with some dict
+      backends that do not support upserting, such as SQL with older SQLite.
 
+   Example:
 
-.. dovecot_plugin:setting:: quota_clone_unset
-   :plugin: quota-clone
-   :values: @boolean
-   :added: 2.4.0,3.0.0
+   .. code-block:: none
 
-   Unset quota information before updating. This is needed with some dict backends
-   that do not support upserting, such as SQL with older SQLite.
+      quota_clone {
+         dict_driver = redis
+         dict_redis_host = 127.0.0.1
+         dict_redis_port = 6379
+         dict_redis_password = testpass
+         unset = yes
+      }
```

Comment:
Since redis doesn't need unset=yes, this is a bit bad example. Also, since Redis is likely not specific to quota_clone, a bit nicer example would be to move the dict_redis_* settings outside the quota_clone {}.

Reply:
I removed unset = yes; however, it is very unclear to me how the dict {} block should be referred from  the quota_clone {} block. I couldn't find examples of that, except for dict_driver = proxy, which has also an accompanying dict_proxy_name

Reply:
That's how it is. The dict {} block is only accessed by the proxy driver. Other drivers access the dict directly, and don't use anything inside dict {}. Although the plan is still to change this further, but it's not quite clear yet how exactly that'll work out.

Reply:
So my intention was simply:

```
dict_redis_host = 127.0.0.1
dict_redis_port = 6379
quota_clone {
  dict_driver = redis
}
```

So in quota_clone it tells to use redis, and the redis configuration is global so anything that uses redis dict won't have to repeat the same redis settings.

Currently dict {} block is using the legacy init code, so if you did want to use it, it would be:

```
dict {
  redisquota = redis:host=127.0.0.1:port=6379
}
quota_clone {
  dict_driver = proxy
  dict_proxy_name = redisquota
}
```

But we'll get rid of this legacy style configuration at some point, so shouldn't use this as an example.